lib/logServer.js

```diff
@@ -10,27 +10,36 @@ const corsResponseHeaders = {
     'Content-Type': 'application/json'
 };
 
-const mochaEndRegex = /# fail ([0-9]+)/;
+const mochaEndRegexSuccess = /# fail ([0]+)/;
+const mochaEndRegexFail = /# fail ([1-9]+)/;
 
 module.exports = (logToConsole = true, runner) =>
     new Promise((resolve, reject) => {
         runner.waitForEvent('log.end');
 
-        const handleLogs = function(logEntry) {
+        const handleLogs = function (logEntry) {
             const logPlatform = logEntry.platform;
             logEntry.logs.forEach(log => {
                 runner.emit('log.data', log, logPlatform);
                 if (logToConsole) {
                     console.log(log);
                 }
 
-                if (mochaEndRegex.test(log)) {
-                    runner.emit('log.end');
+                if (mochaEndRegexSuccess.test(log)) {
```

Comment:
The only difference if whether a test failed is the payload of the 'log.end' event. The bodies of these two if statements can just set the payload of the emitted event.

Since here we are defining an API for all event-based tasks for the runner, it would be better to conform to Nodejs convention - first argument should be an error and the second would be the result. So something like

```javascript
let err = null;
if (mochaEndRegexFail.test(log)) {
  err = new Error('One or more tests failed');
}
runner.emit('log.end', err);
```